x/mongo/driver/topology/topology.go

```diff
@@ -605,7 +617,11 @@ func (t *Topology) apply(ctx context.Context, desc description.Server) descripti
 	var current description.Topology
 	var err error
 	current, desc, err = t.fsm.apply(desc)
+
 	if err != nil {
+		updatedDesc := t.Description()
```

Comment:
This assumes that `apply` will only return a non-nil error if there was a compatibility error. This may be the case right now, but it could change in the future and I'd prefer to not rely on it. Can you think of a way to have `fsm.apply()` return the right data and still have this function store all of the information it needs?